DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1

```diff
@@ -88,62 +137,73 @@ function Set-TargetResource
         $CheckSum = 'ModifiedDate'
     )
 
-    if (-not (Test-Path $VhdPath)) { throw "Specified destination path $VhdPath does not exist!"}   
+    if(-not (Test-Path -Path $VhdPath))
+    {
+        Write-Verbose ($localizedData.VHDMissing -f $VhdPath)
+        break;
+    }
     
     # mount the VHD.
-    $mountedVHD = EnsureVHDState -Mounted -vhdPath $VhdPath
+    $mountVHD = EnsureVHDState -Mounted -vhdPath $VhdPath
 
     try
     {
-            # show the drive letters.
-            Get-PSDrive | Write-Verbose
-
-            $mountedDrive = $mountedVHD | Get-Disk | Get-Partition | Get-Volume
+        # Retrieve the current drive letter for the VHD
+        if($VhdPartitionNumber)
```

Comment:
Looks like there is some common code that can be put in a separate function and shared across test/get/set. For e.g. getting $mountedDrive